docker/Dockerfile

```diff
@@ -1,4 +1,20 @@
-# Inherit from the Drupal 9 image on Docker Hub.
+# Use PHP 8 image to build GEOS PHP extension.
+FROM php:8.1-cli as geos
```

Comment:
Why did you prefer to use `php:8.1-cli` image?
I don't see the downside of using `drupal:9.5`, and whenever a Drupal with a new version of PHP is required, the image definition here doesn't need to be updated.
Maybe it would be more clear when to see with the multi-stages Dockerfile for multiple environments that I'm opening the PR soon.

Reply:
It just felt like the right thing to do since building GEOS itself does not require any of the additional complexity included in the `drupal` image layers. We could potentially change it to `php:8-cli` (instead of `8.1`), so that we don't need to update it again (until PHP 9 + whatever future Drupal version that supports that). I doubt that it matters if GEOS is built with `8.1` vs `8.2` vs `8.x`... I imagine it would work the same when we copy it into the `drupal` image layer. But maybe that's not correct and there is logic in the GEOS build that varies based on minor PHP version.

For what it's worth, Drupal 9 only works with PHP 8.1, and is only supported until the end of this month. We are working on farmOS 3.x on Drupal 10 right now, which uses PHP 8.2 as the basis of the `drupal:10.1` image. Drupal itself does not support PHP 8.3 yet, although that is "planned" for Drupal 10.2.

https://www.drupal.org/docs/getting-started/system-requirements/php-requirements

This is just to say: we are already quite intentional about the PHP version that we use in these images. It doesn't change often, and when it does we also need to update other things. So it won't be adding much additional maintenance to use the `php` image for this IMO.

All that said, I don't feel very strongly about this, so if there is agreement that it is simple to use the same `drupal` image for both stages, I'm fine with that too.

Reply:
You are right that it doesn't require anything else than what belongs to PHP official images, although the layers that Drupal is adding to the image are not affecting the final farmOS image. 
I'm still thinking that it's better to keep Drupal image. As you mention, the changes will be required anyway, but reducing the consistent number of changes also keeps down the unaware errors.
Let's imagine that Drupal decides to compile one of the PHP modules with some specific arguments that are not available on PHP image, and that we keep the line 30:
```suggestion
COPY --from=geos /usr/local/lib/php/extensions/ /usr/local/lib/php/extensions/
```
That would mean that the module X, will be overwritten with the PHP version, without the specified argument, and might lead to a debugging time until the cause is found. 

This could be solved by coping only the `geo.so` file, yet the extension's directory has inside a folder that ends with the PHP extension value, which I don't know when that is it updated. An alternative is to add scripting to find the right path.
![Screenshot from 2023-10-21 14-52-15](https://github.com/farmOS/farmOS/assets/232739/d189bb23-6587-48d7-b134-316266a8828e)
![Screenshot from 2023-10-21 14-52-41](https://github.com/farmOS/farmOS/assets/232739/859a1a0c-246b-4099-ae0d-abc4664f7e61)
 
It might be an overthinking by ignorance, but those are the possible scenarios that I can imagine.
In my opinion, keeping the base image prevents all of them.

Reply:
> I'm still thinking that it's better to keep Drupal image. As you mention, the changes will be required anyway, but reducing the consistent number of changes also keeps down the unaware errors.

OK we can keep the Drupal image. :-)

I'll change it back and force push.